Source/Flow/Public/Interfaces/FlowDataPinValueSupplierInterface.h

```diff
@@ -0,0 +1,88 @@
+// Copyright https://github.com/MothCocoon/FlowGraph/graphs/contributors
+
+#pragma once
+
+#include "Types/FlowDataPinResults.h"
+
+#include "UObject/Interface.h"
+
+#include "FlowDataPinValueSupplierInterface.generated.h"
+
+// Interface to define a Flow Data Pin value supplier.  This is generally a UFlowNode subclass, 
+// but we may support external suppliers that are not flow nodes in the future 
+// (eg, for supplying configuration values for the root graph)
+UINTERFACE(MinimalAPI, Blueprintable, DisplayName = "Flow Data Pin Value Supplier Interface")
+class UFlowDataPinValueSupplierInterface : public UInterface
+{
+	GENERATED_BODY()
+};
+
+class FLOW_API IFlowDataPinValueSupplierInterface
+{
+	GENERATED_BODY()
+
+public:
+	// Can this node actually supply Data Pin values?
+	// Implementers of this interface will need to use their own logic to answer this question.
+	UFUNCTION(BlueprintNativeEvent, Category = "FlowNode", DisplayName = "Can Supply DataPin Values")
+	bool CanSupplyDataPinValues() const;
+	virtual bool CanSupplyDataPinValues_Implementation() const { return false; }
+
+	// Must implement TrySupplyDataAs... for every EFlowPinType
+	FLOW_ASSERT_ENUM_MAX(EFlowPinType, 12);
+
+	// Try to supply the value for a data Bool pin
+	UFUNCTION(BlueprintNativeEvent, Category = "FlowNode", DisplayName = "Try Supply DataPin As Bool")
```

Comment:
Each different type of data pin requires its own supply value function signature.

These are blueprint native, so that blueprint objects can implement these function(s) as well.